In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [13]:
### this is a comment 

In [41]:
accre_data = pd.read_csv(r'C:\Users\chris\data_science_projects\data-question-2-dream-crushers\july-2017-thru-june-2018.txt', delimiter='|')
accre_data.head(n=20)

C:\Users\chris\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2785: DtypeWarning: Columns (0,20) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


JobID  JobIDRaw Cluster   Partition      Account        Group    GID  \
0   15806649  15806649   accre  production      h_vuiis      h_vuiis  36052   
1   15806664  15806664   accre  production      h_vuiis      h_vuiis  36052   
2   15806670  15806670   accre  production      h_vuiis      h_vuiis  36052   
3   15806671  15806671   accre  production      h_vuiis      h_vuiis  36052   
4   15811617  15811617   accre  production  cms_samtest  cms_samtest  59297   
5   15811618  15811618   accre  production   cms_stage2   cms_stage1  59298   
6   15811619  15811619   accre  production   cms_stage2          cms  31000   
7   15811621  15811621   accre  production  cms_samtest  cms_samtest  59297   
8   15811624  15811624   accre  production   cms_stage2   cms_stage1  59298   
9   15811625  15811625   accre  production   cms_stage2          cms  31000   
10  15811628  15811628   accre  production      h_vuiis      h_vuiis  36052   
11  15811629  15811629   accre  production      h_vuiis      h_vuiis  36052   
12  15811630  15811630   accre  production      h_vuiis      h_vuiis  36052   
13  15811631  15811631   accre  production      h_vuiis      h_vuiis  36052   
14  15811632  15811632   accre  production      h_vuiis      h_vuiis  36052   
15  15811633  15811633   accre  production      h_vuiis      h_vuiis  36052   
16  15811634  15811634   accre  production      h_vuiis      h_vuiis  36052   
17  15811635  15811635   accre  production      h_vuiis      h_vuiis  36052   
18  15811636  15811636   accre  production      h_vuiis      h_vuiis  36052   
19  15811637  15811637   accre  production      h_vuiis      h_vuiis  36052   

          User       UID               Submit  \
0   masispider       NaN  2017-07-02T00:40:58   
1   masispider       NaN  2017-07-02T00:40:58   
2   masispider       NaN  2017-07-02T00:40:58   
3   masispider       NaN  2017-07-02T00:40:58   
4     uscms010       NaN  2017-07-01T05:00:00   
5      autocms    9209.0  2017-07-01T05:00:01   
6         tuos  126986.0  2017-07-01T05:00:02   
7     uscms010       NaN  2017-07-01T05:00:22   
8      autocms    9209.0  2017-07-01T05:01:01   
9         tuos  126986.0  2017-07-01T05:01:02   
10    vuiiscci       NaN  2017-07-01T05:01:22   
11    vuiiscci       NaN  2017-07-01T05:01:22   
12    vuiiscci       NaN  2017-07-01T05:01:23   
13    vuiiscci       NaN  2017-07-01T05:01:23   
14    vuiiscci       NaN  2017-07-01T05:01:23   
15    vuiiscci       NaN  2017-07-01T05:01:23   
16    vuiiscci       NaN  2017-07-01T05:01:23   
17    vuiiscci       NaN  2017-07-01T05:01:24   
18    vuiiscci       NaN  2017-07-01T05:01:24   
19    vuiiscci       NaN  2017-07-01T05:01:24   

                          ...                              State NNodes NCPUS  \
0                         ...                          COMPLETED      1     2   
1                         ...                          COMPLETED      1     2   
2                         ...                          COMPLETED      1     2   
3                         ...                          COMPLETED      1     2   
4                         ...                          COMPLETED      1     1   
5                         ...                          COMPLETED      1     1   
6                         ...                          COMPLETED      1     1   
7                         ...                          COMPLETED      1     1   
8                         ...                          COMPLETED      1     1   
9                         ...                          COMPLETED      1     1   
10                        ...                          COMPLETED      1     1   
11                        ...                          COMPLETED      1     1   
12                        ...                          COMPLETED      1     1   
13                        ...                          COMPLETED      1     1   
14                        ...                          COMPLETED      1     1   
15                   

In [42]:
accre_data['State'].value_counts()

COMPLETED              7150200
CANCELLED by 29607     3354734
FAILED                  902907
CANCELLED by 29605      355327
TIMEOUT                 160402
CANCELLED by 112888     105801
CANCELLED by 233674      66809
CANCELLED by 207302      44886
CANCELLED by 0           29569
CANCELLED by 551122      24607
CANCELLED by 53988       19998
CANCELLED by 233240      17531
CANCELLED by 237602      17013
CANCELLED by 342713      16950
CANCELLED by 501421      16363
CANCELLED by 29789       14925
PREEMPTED                13425
CANCELLED by 144605      12054
CANCELLED by 169859      11882
CANCELLED by 342288      10227
CANCELLED by 497968       9284
CANCELLED by 200524       7099
CANCELLED by 171302       6349
CANCELLED by 454080       5956
CANCELLED by 115292       5643
CANCELLED by 154990       5162
CANCELLED by 664085       5073
CANCELLED by 59223        4985
CANCELLED by 229206       4860
CANCELLED by 237813       4624
                        ...   
CANCELLED by 645144          2
CANCELLE

In [91]:
Job_State=accre_data['State']

In [92]:
print(Job_State)

0           COMPLETED
1           COMPLETED
2           COMPLETED
3           COMPLETED
4           COMPLETED
5           COMPLETED
6           COMPLETED
7           COMPLETED
8           COMPLETED
9           COMPLETED
10          COMPLETED
11          COMPLETED
12          COMPLETED
13          COMPLETED
14          COMPLETED
15          COMPLETED
16            TIMEOUT
17          COMPLETED
18          COMPLETED
19          COMPLETED
20          COMPLETED
21          COMPLETED
22          COMPLETED
23          COMPLETED
24          COMPLETED
25          COMPLETED
26            TIMEOUT
27          COMPLETED
28          COMPLETED
29          COMPLETED
              ...    
12499359       FAILED
12499360    COMPLETED
12499361       FAILED
12499362       FAILED
12499363    COMPLETED
12499364    COMPLETED
12499365    COMPLETED
12499366    COMPLETED
12499367    COMPLETED
12499368    COMPLETED
12499369    COMPLETED
12499370    COMPLETED
12499371    COMPLETED
12499372    COMPLETED
12499373  

# Calculating the proportion of failed jobs to total jobs

In [99]:
total_failures = 902907

In [103]:
print(total_failures / len(Job_State))

0.0722360909001232


# About 7% of all jobs fail (902,907 out of 12,499,389 total jobs failed).

# Central Question: Are these 7% of failures coming from a disproportionate number of particular nodes?

# Before we answer that question, we will only consider jobs that used 1 node (which 98.97% of all jobs).

In [107]:
accre_data['NNodes'].value_counts()

1      12370952
0        108668
2          9344
3          2746
4          2357
5           811
100         741
6           497
8           488
36          273
50          237
7           202
10          157
82          144
16          131
25          122
64          112
12          102
32           82
9            73
20           62
13           61
11           60
15           56
14           52
17           45
19           45
18           42
30           37
128          32
         ...   
96            2
61            2
76            2
98            2
83            2
77            1
89            1
188           1
136           1
132           1
131           1
125           1
124           1
118           1
117           1
116           1
115           1
114           1
111           1
109           1
84            1
85            1
104           1
86            1
97            1
200           1
95            1
88            1
93            1
94            1
Name: NNodes, Length: 12

In [113]:
One_Node= accre_data[accre_data['NNodes'] == 1]

In [114]:
print(One_Node)

             JobID  JobIDRaw Cluster   Partition      Account        Group  \
0         15806649  15806649   accre  production      h_vuiis      h_vuiis   
1         15806664  15806664   accre  production      h_vuiis      h_vuiis   
2         15806670  15806670   accre  production      h_vuiis      h_vuiis   
3         15806671  15806671   accre  production      h_vuiis      h_vuiis   
4         15811617  15811617   accre  production  cms_samtest  cms_samtest   
5         15811618  15811618   accre  production   cms_stage2   cms_stage1   
6         15811619  15811619   accre  production   cms_stage2          cms   
7         15811621  15811621   accre  production  cms_samtest  cms_samtest   
8         15811624  15811624   accre  production   cms_stage2   cms_stage1   
9         15811625  15811625   accre  production   cms_stage2          cms   
10        15811628  15811628   accre  production      h_vuiis      h_vuiis   
11        15811629  15811629   accre  production      h_vuiis   

In [115]:
accre_data['NodeList'].value_counts()

None assigned                                                                                                                                                                                                                                                                                                                                                                                                                             3470227
vmp1409                                                                                                                                                                                                                                                                                                                                                                                                                                     92419
vmp1242                                                                                                                             

In [123]:
accre_data[(accre_data['NodeList'] =="vmp1409") | (accre_data['State'] == "FAILED")]

JobID  JobIDRaw Cluster   Partition        Account  \
386       15812080  15812080   accre  production  segrest_group   
441       15812174  15812174   accre  production  segrest_group   
453       15812190  15812190   accre  production  segrest_group   
537       15812298  15812298   accre  production  segrest_group   
543       15812304  15812304   accre  production        h_vuiis   
544       15812305  15812305   accre  production        h_vuiis   
545       15812306  15812306   accre  production        h_vuiis   
546       15812307  15812307   accre  production        h_vuiis   
547       15812308  15812308   accre  production        h_vuiis   
548       15812309  15812309   accre  production        h_vuiis   
549       15812310  15812310   accre  production        h_vuiis   
576       15812351  15812351   accre  production  segrest_group   
718       15812558  15812558   accre  production        h_vuiis   
719       15812559  15812559   accre  production        h_vuiis   
720       15812560  15812560   accre  production        h_vuiis   
721       15812561  15812561   accre  production        h_vuiis   
722       15812562  15812562   accre  production        h_vuiis   
723       15812563  15812563   accre  production        h_vuiis   
724       15812564  15812564   accre  production        h_vuiis   
725       15812565  15812565   accre  production        h_vuiis   
726       15812566  15812566   accre  production        h_vuiis   
727       15812567  15812567   accre  production        h_vuiis   
728       15812568  15812568   accre  production        h_vuiis   
729       15812569  15812569   accre  production        h_vuiis   
730       15812570  15812570   accre  production        h_vuiis   
731       15812571  15812571   accre  production        h_vuiis   
732       15812572  15812572   accre  production        h_vuiis   
733       15812573  15812573   accre  production        h_vuiis   
734       15812574  15812574   accre  production        h_vuiis   
735       15812575  15812575   accre  production        h_vuiis   
...            ...       ...     ...         ...            ...   
12498761  28330312  28330312   accre  production        h_vuiis   
12498840  28330429  28330429   accre  production    cms_lowprio   
12498940  28330559  28330559   accre  production    cms_lowprio   
12498998  28330638  28330638   accre  production    cms_lowprio   
12499036  28330680  28330680   accre  production     cms_stage3   
12499037  28330681  28330681   accre  production     cms_stage3   
12499038  28330682  28330682   accre  production     cms_stage3   
12499039  28330683  28330683   accre  production     cms_stage3   
12499040  28330684  28330684   accre  production     cms_stage3   
12499041  28330685  28330685   accre  production     cms_stage3   
12499043  28330687  28330687   accre  production     cms_stage3   
12499044  28330688  28330688   accre  production     cms_stage3   
12499045  28330689  28330689   accre  production     cms_stage3   
12499046  28330690  28330690   accre  production     cms_stage3   
12499047  28330691  28330691   accre  production     cms_stage3   
12499093  28330752  28330752   accre  production    cms_lowprio   
12499246  28331043  28331043   accre  production        h_vuiis   
12499259  28331073  28331073   accre  production     cms_stage3   
12499260  28331074  28331074   accre  production     cms_stage3   
12499287  28331105  28331105   accre  production     cms_stage3   
12499288  28331106  28331106   accre  production     cms_stage3   
12499289  28331107  28331107   accre  production     cms_stage3   
12499359  28331320  28331320   accre  production     cms_stage3   
12499361  28331322  28331322   accre  production     cms_stage3   
12499362  28331323  28331323   accre  production     cms_stage3   
12499382  28331349  28331349   accre  production     cms_stage3   
12499383  28331350  28331350   accre  production     cms_stage3   
12499384  28331351  28331351   accre  production     cms_sta

# Let's look @ at the failure rate of the most used node--vmp1409--before we look at all nodes. 

In [128]:
accre_data[(accre_data['State'] == "FAILED") & (accre_data["NodeList"] == "vmp1409")]

JobID  JobIDRaw Cluster   Partition            Account  \
3151714      18971910_0  18971911   accre  production              accre   
3151715      18971910_1  18971912   accre  production              accre   
3151716      18971910_2  18971913   accre  production              accre   
3151717      18971910_3  18971914   accre  production              accre   
3151718      18971910_4  18971915   accre  production              accre   
3151719      18971910_5  18971916   accre  production              accre   
3151720      18971910_6  18971917   accre  production              accre   
3151721      18971910_7  18971918   accre  production              accre   
3151722      18971910_8  18971919   accre  production              accre   
3151723      18971910_9  18971920   accre  production              accre   
3151724     18971910_10  18971921   accre  production              accre   
3151725     18971910_11  18971922   accre  production              accre   
3151726     18971910_12  18971923   accre  production              accre   
3151727     18971910_13  18971924   accre  production              accre   
3151728     18971910_14  18971925   accre  production              accre   
3151729     18971910_15  18971926   accre  production              accre   
3740339        19569113  19569113   accre  production               sbcs   
3753323   19517743_2070  19582097   accre  production               lola   
3753324   19517743_2071  19582098   accre  production               lola   
3753326   19517743_2073  19582100   accre  production               lola   
3819468        19648242  19648242   accre  production                cms   
3836378    19656557_111  19665152   accre  production  h_biostat_student   
3837250    19656557_169  19666024   accre  production  h_biostat_student   
3838666     19656560_65  19667440   accre  production  h_biostat_student   
3838667     19656560_66  19667441   accre  production  h_biostat_student   
3840344     19656561_62  19669118   accre  production  h_biostat_student   
3840345     19656561_63  19669119   accre  production  h_biostat_student   
3855039    19677250_225  19683813   accre  production  h_biostat_student   
3948452    19776978_117  19777226   accre  production           beam_lab   
4044901        19873678  19873678   accre  production      capra_lab_csb   
...                 ...       ...     ...         ...                ...   
12433387   28262469_144  28262674   accre  production          capra_lab   
12433388   28262469_145  28262675   accre  production          capra_lab   
12433389   28262469_146  28262676   accre  production          capra_lab   
12433607   28262469_230  28262894   accre  production          capra_lab   
12433608   28262469_231  28262895   accre  production          capra_lab   
12433609   28262469_232  28262896   accre  production          capra_lab   
12433943   28262469_356  28263230   accre  production          capra_lab   
12433944   28262469_357  28263231   accre  production          capra_lab   
12433945   28262469_358  28263232   accre  production          capra_lab   
12438073   28262975_611  28267360   accre  production          capra_lab   
12440761    28269597_92  28270048   accre  production          capra_lab   
12441499   28269599_153  28270786   accre  production          capra_lab   
12441587   28269599_194  28270874   accre  production          capra_lab   
12441747   28269599_273  28271034   accre  production          capra_lab   
12441846   28269599_321  28271133   accre  production          capra_lab   
12441942   28269599_371  28271229   accre  production          capra_lab   
12442043   28269599_425  28271330   accre  production          capra_lab   
12442129   28269599_464  28271416   accre  production          capra_lab   
12460969    28290260_10  28290260   accre  production      p_meiler_casp   
12460979     28290260_9  28290270   accre  production      p_meiler_casp   
12461035     28290321_3  28290326   accre  production      p_meile

# Node vmp1409 was used to execute the most jobs: 3,470,227 jobs. It had 1,119 failures. The frequency of failure was less than .03%--far below the average fail rate. Because this node executed a sizeable percentage of all jobs and had such a low failure rate, I predict that there is a node with a disproportionately high failure rate that is skewing the overall average.

# Let's keep analyzing each node.

In [130]:
accre_data[(accre_data['State'] == "FAILED") & (accre_data["NodeList"] == "vmp1242")]

JobID  JobIDRaw Cluster   Partition            Account  \
40474         15857888  15857888   accre  production        cms_samtest   
80967         15898381  15898381   accre  production        cms_samtest   
80968         15898382  15898382   accre  production        cms_samtest   
82287         15899701  15899701   accre  production        cms_samtest   
82288         15899702  15899702   accre  production        cms_samtest   
82292         15899706  15899706   accre  production        cms_samtest   
82293         15899707  15899707   accre  production        cms_samtest   
82294         15899708  15899708   accre  production        cms_samtest   
82296         15899710  15899710   accre  production        cms_samtest   
100900        15919242  15919242   accre  production        cms_samtest   
100912        15919254  15919254   accre  production        cms_samtest   
105080        15923422  15923422   accre  production        cms_samtest   
105087        15923429  15923429   accre  production        cms_samtest   
106978        15925320  15925320   accre  production        cms_samtest   
106979        15925321  15925321   accre  production        cms_samtest   
106980        15925322  15925322   accre  production        cms_samtest   
106981        15925323  15925323   accre  production        cms_samtest   
110607        15928949  15928949   accre  production        cms_samtest   
110611        15928953  15928953   accre  production        cms_samtest   
110617        15928959  15928959   accre  production        cms_samtest   
110700        15929042  15929042   accre  production        cms_samtest   
111983        15930325  15930325   accre  production        cms_samtest   
111988        15930330  15930330   accre  production        cms_samtest   
112450        15930792  15930792   accre  production        cms_samtest   
112454        15930796  15930796   accre  production        cms_samtest   
112456        15930798  15930798   accre  production        cms_samtest   
112457        15930799  15930799   accre  production        cms_samtest   
114036        15932378  15932378   accre  production        cms_samtest   
114075        15932417  15932417   accre  production        cms_samtest   
114122        15932464  15932464   accre  production        cms_samtest   
...                ...       ...     ...         ...                ...   
12054558   27883206_85  27883291   accre  production           beam_lab   
12054559   27883206_86  27883292   accre  production           beam_lab   
12054560   27883206_87  27883293   accre  production           beam_lab   
12054561   27883206_88  27883294   accre  production           beam_lab   
12061453      27890186  27890186   accre  production            h_vuiis   
12063907  27892444_178  27892640   accre  production         csb_chazin   
12063908  27892444_179  27892641   accre  production         csb_chazin   
12063909  27892444_180  27892642   accre  production         csb_chazin   
12063910  27892444_181  27892643   accre  production         csb_chazin   
12064183  27892445_143  27892916   accre  production         csb_chazin   
12064184  27892445_144  27892917   accre  production         csb_chazin   
12064185  27892445_145  27892918   accre  production         csb_chazin   
12064186  27892445_146  27892919   accre  production         csb_chazin   
12121264   27938931_52  27950000   accre  production          p_vantage   
12121451   27938931_78  27950187   accre  production          p_vantage   
12145024      27973760  27973760   accre  production                cms   
12198397    28027126_6  28027134   accre  production          p_vantage   
12198443    28027172_7  28027180   accre  production          p_vantage   
12218936      28047673  28047673   accre  production          econgrads   
12228322  28056253_104  28057060   accre  production           beam_lab   
12262628  28087254_146  28091368   accre  production  h_biostat_student   
12265342  28087254_338  28094082   accre  

# Node 1242 has a failure rate of 5.64%.

In [131]:
accre_data[(accre_data['State'] == "FAILED") & (accre_data["NodeList"] == "vmp1257")]

JobID  JobIDRaw Cluster   Partition            Account  \
21344       15838734_3  15838737   accre  production  h_biostat_student   
21372       15838762_9  15838771   accre  production  h_biostat_student   
21399       15838800_5  15838805   accre  production  h_biostat_student   
21428       15838839_2  15838842   accre  production  h_biostat_student   
21473       15838882_5  15838887   accre  production  h_biostat_student   
21538       15838947_5  15838952   accre  production  h_biostat_student   
28425       15845816_9  15845839   accre  production       p_csb_meiler   
28426      15845816_10  15845840   accre  production       p_csb_meiler   
28427      15845816_11  15845841   accre  production       p_csb_meiler   
28572         15845986  15845986   accre  production      capra_lab_csb   
28573         15845987  15845987   accre  production      capra_lab_csb   
28574         15845988  15845988   accre  production      capra_lab_csb   
46043         15863457  15863457   accre  production                cms   
46044         15863458  15863458   accre  production                cms   
46048         15863462  15863462   accre  production                cms   
46049         15863463  15863463   accre  production                cms   
46050         15863464  15863464   accre  production                cms   
46051         15863465  15863465   accre  production                cms   
46052         15863466  15863466   accre  production                cms   
46053         15863467  15863467   accre  production                cms   
46055         15863469  15863469   accre  production                cms   
50936         15868350  15868350   accre  production  h_biostat_student   
50938         15868352  15868352   accre  production  h_biostat_student   
60519         15877933  15877933   accre  production  h_biostat_student   
60520         15877934  15877934   accre  production  h_biostat_student   
60521         15877935  15877935   accre  production  h_biostat_student   
60522         15877936  15877936   accre  production  h_biostat_student   
60523         15877937  15877937   accre  production  h_biostat_student   
60524         15877938  15877938   accre  production  h_biostat_student   
61307         15878721  15878721   accre  production  h_biostat_student   
...                ...       ...     ...         ...                ...   
12230532      28059270  28059270   accre  production                cgg   
12244998      28073736  28073736   accre  production            h_vuiis   
12249352      28078091  28078091   accre  production                cms   
12249354      28078093  28078093   accre  production                cms   
12267362   28096073_29  28096102   accre  production          capra_lab   
12267466  28096073_133  28096206   accre  production          capra_lab   
12267509  28096073_176  28096249   accre  production          capra_lab   
12267613  28096073_279  28096353   accre  production          capra_lab   
12267650  28096073_316  28096390   accre  production          capra_lab   
12267716  28096073_382  28096456   accre  production          capra_lab   
12267778  28096073_444  28096518   accre  production          capra_lab   
12267827  28096073_493  28096567   accre  production          capra_lab   
12267896  28096073_562  28096636   accre  production          capra_lab   
12267944  28096073_610  28096684   accre  production          capra_lab   
12267980  28096073_646  28096720   accre  production          capra_lab   
12268053  28096073_719  28096793   accre  production          capra_lab   
12268089  28096073_755  28096829   accre  production          capra_lab   
12372836  28174984_522  28202076   accre  production               lola   
12395487      28224732  28224732   accre  production             coxlab   
12395488      28224733  28224733   accre  production             coxlab   
12395494      28224739  28224739   accre  production             coxlab   
12395496      28224741  28224741   accre  

# Node1257 has a failure rate of 1.66%.

In [133]:
accre_data[(accre_data['State'] == "FAILED") & (accre_data["NodeList"] == "vmp1001")]

JobID  JobIDRaw Cluster   Partition            Account  \
22735          15840149  15840149   accre  production           h_taylor   
26631          15844045  15844045   accre  production  h_biostat_student   
26650          15844064  15844064   accre  production  h_biostat_student   
26669          15844083  15844083   accre  production  h_biostat_student   
26688          15844102  15844102   accre  production  h_biostat_student   
27021          15844435  15844435   accre  production  h_biostat_student   
27421          15844835  15844835   accre  production  h_biostat_student   
85401          15903743  15903743   accre  production            h_vuiis   
88843          15907185  15907185   accre  production            h_vuiis   
110662         15929004  15929004   accre  production         cms_stage2   
110922         15929264  15929264   accre  production         cms_stage2   
110980         15929322  15929322   accre  production         cms_stage2   
111011         15929353  15929353   accre  production         cms_stage2   
119518         15937860  15937860   accre  production        cms_lowprio   
141740         15960082  15960082   accre  production         cms_stage2   
176898         15995240  15995240   accre  production          mass_spec   
189353     16004885_748  16007695   accre  production       p_csb_meiler   
189426     16004885_771  16007768   accre  production       p_csb_meiler   
189427     16004885_772  16007769   accre  production       p_csb_meiler   
196219         16014561  16014561   accre  production  h_biostat_student   
206175         16024517  16024517   accre  production  h_biostat_student   
206176         16024518  16024518   accre  production  h_biostat_student   
206177         16024519  16024519   accre  production  h_biostat_student   
227417    16045626_1127  16045759   accre  production           mip_eecs   
227418    16045626_1128  16045760   accre  production           mip_eecs   
228801         16047143  16047143   accre  production                cms   
228872         16047214  16047214   accre  production                cms   
229289         16047631  16047631   accre  production                cms   
229424         16047766  16047766   accre  production                cms   
239383         16057725  16057725   accre  production                cms   
...                 ...       ...     ...         ...                ...   
11231911       27060632  27060632   accre  production        cms_samtest   
11231978       27060699  27060699   accre  production        cms_samtest   
11232376       27061097  27061097   accre  production        cms_samtest   
11232572       27061293  27061293   accre  production                cms   
11232678       27061399  27061399   accre  production        cms_samtest   
11232685       27061406  27061406   accre  production        cms_samtest   
11232687       27061408  27061408   accre  production        cms_samtest   
11233175       27061896  27061896   accre  production                cms   
11233176       27061897  27061897   accre  production                cms   
11233178       27061899  27061899   accre  production                cms   
11233179       27061900  27061900   accre  production                cms   
11373966       27202687  27202687   accre  production      segrest_group   
11373988       27202709  27202709   accre  production      segrest_group   
11435409       27264132  27264132   accre  production              h_cqs   
11462162    27290069_13  27290885   accre  production       anderson_mri   
11499706     27328425_4  27328429   accre  production    biostat_faculty   
11512485       27341209  27341209   accre  production    biostat_faculty   
11580730       27409454  27409454   accre  production        cms_lowprio   
11671128       27499855  27499855   accre  production            h_vuiis   
11671129       27499856  27499856   accre  production            h_vuiis   
11671191       27499918  27499918   accre  production            h

In [ ]:
# Commentary

In [132]:
accre_data[(accre_data['State'] == "FAILED") & (accre_data["NodeList"] == "vmp1298")]

JobID  JobIDRaw Cluster   Partition       Account  \
37430        15854844  15854844   accre  production      rokaslab   
37623        15855037  15855037   accre  production      rokaslab   
231147       16049489  16049489   accre  production    biostat_it   
231200       16049542  16049542   accre  production    biostat_it   
231726       16050068  16050068   accre  production          sbcs   
372114    16185863_38  16190456   accre  production  p_csb_meiler   
586793       16405136  16405136   accre  production           cms   
586815       16405158  16405158   accre  production           cms   
647517       16465861  16465861   accre  production       h_vuiis   
647779       16466123  16466123   accre  production       h_vuiis   
668769       16487113  16487113   accre  production     capra_lab   
668797       16487141  16487141   accre  production     capra_lab   
668858       16487202  16487202   accre  production     capra_lab   
722602       16540947  16540947   accre  production     capra_lab   
722642       16540987  16540987   accre  production     capra_lab   
722922       16541267  16541267   accre  production     capra_lab   
723492       16541837  16541837   accre  production     capra_lab   
894410       16712757  16712757   accre  production           cms   
894411       16712758  16712758   accre  production           cms   
894413       16712760  16712760   accre  production           cms   
895070       16713417  16713417   accre  production       h_vuiis   
895131       16713478  16713478   accre  production       h_vuiis   
895205       16713552  16713552   accre  production       h_vuiis   
895304       16713651  16713651   accre  production       h_vuiis   
895939       16714286  16714286   accre  production           cms   
895949       16714296  16714296   accre  production           cms   
895955       16714302  16714302   accre  production           cms   
896422       16714769  16714769   accre  production           cms   
896428       16714775  16714775   accre  production           cms   
896452       16714799  16714799   accre  production           cms   
...               ...       ...     ...         ...           ...   
10529355     26358613  26358613   accre  production   cms_samtest   
10529370     26358628  26358628   accre  production   cms_samtest   
10533016     26362274  26362274   accre  production   cms_samtest   
10533017     26362275  26362275   accre  production   cms_samtest   
10533018     26362276  26362276   accre  production   cms_samtest   
10533019     26362277  26362277   accre  production   cms_samtest   
10562374     26391632  26391632   accre  production   cms_samtest   
10863464     26692731  26692731   accre  production   cms_samtest   
10863510     26692777  26692777   accre  production   cms_samtest   
10863762     26693029  26693029   accre  production   cms_samtest   
11403921     27232643  27232643   accre  production   cms_samtest   
11542680     27371404  27371404   accre  production   cms_samtest   
11627977     27456702  27456702   accre  production   cms_samtest   
12105977     27934713  27934713   accre  production   cms_samtest   
12106004     27934740  27934740   accre  production   cms_samtest   
12201966     28030703  28030703   accre  production   cms_samtest   
12247321     28076060  28076060   accre  production   cms_samtest   
12281048     28109788  28109788   accre  production   cms_samtest   
12465342     28294634  28294634   accre  production   cms_samtest   
12465590     28294884  28294884   accre  production   cms_samtest   
12482602     28312559  28312559   accre  production   cms_samtest   
12482672     28312629  28312629   accre  production   cms_samtest   
12482779     28312736  28312736   accre  production   cms_samtest   
12482780     28312737  28312737   accre  production   cms_samtest   
12485481     28315513  28315513   accre  production   cms_samtest   
12485535     28315568  28315568   accre  production   cms_samtest   
12485547     2